### PR TITLE
(#48, #45) Sink Dashboard

### LINKED ISSUE #45: Api Service for Sink Management
- Configure API Service for Sink Management in Mock Server
- Integrate Mock endpoint with Mainflux
- Implement Angular Sink service

https://github.com/ns1labs/orb/wiki/Architecture:-Sinks

### PR DESCRIPTION
**Pull request by gpazuch**
Add first iteration of the sink management dashboard.

* sink dashboard in usable state for testing
 - Name, Description, Status, Tags and Action Buttons
* sink service api integration
* sink add modal(temporary) to enable creation of sinks
* sink view and delete modals are just placeholders - not usable yet
* sink filter is placeholder and non functional - late design change.
* early table version, no column ordering or anything fancy

resolves #45, #48

do some of the boilerplate work of routing and templates of tasks #50 #46 #49

### CODE REVIEW

File: ui/src/app/@theme/components/header/header.component.html [hunk 1]
```diff
@@ -1,28 +1,29 @@
+<script src="../../../../assets/env.js"></script>
```
Comment (daitan-rovaris) [inline ui/src/app/@theme/components/header/header.component.html]: Check if this is really necessary

File: ui/src/app/@theme/styles/orb/_orb-overrides.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
+@mixin orb-overrides() {
+  @include nb-for-themes(orb-dark) {
+    nb-layout-header {
+      nb-actions, .logo-container {
+        nb-icon, .user-name {
+          color: nb-theme(color-basic-100) !important;
```
[the remark this answers is not in the document]
Comment (daitan-rovaris) [reply ui/src/app/@theme/styles/orb/_orb-overrides.scss:6]: referring to the important notation

File: ui/src/app/common/services/sinks/sinks.service.ts [hunk 3]
```diff
@@ -1,122 +1,120 @@
...
-import { Sink } from 'app/common/interfaces/sink.interface';
-import { NotificationsService } from 'app/common/services/notifications/notifications.service';
-import { PageFilters } from 'app/common/interfaces/mainflux.interface';
+import {environment} from 'environments/environment';
+import {Sink} from 'app/common/interfaces/sink.interface';
+import {NotificationsService} from 'app/common/services/notifications/notifications.service';
+import {PageFilters} from 'app/common/interfaces/mainflux.interface';
 
+// default filters
 const defLimit: number = 20;
```
Comment (daitan-rovaris) [inline ui/src/app/common/services/sinks/sinks.service.ts:12]: Constants UPPERCASE

File: ui/src/app/pages/agents/agents.component.html [hunk 4]
```diff
@@ -36,7 +36,7 @@
     <ngx-table-component [config]="tableConfig"
                          [page]="page"
                          (editEvent)=openEditModal($event)
-                         (delEvent)=openDeleteModal($event)
+                         (deleteEvent)="openDeleteModal($event)"
```
Comment (daitan-rovaris) [inline ui/src/app/pages/agents/agents.component.html]: is this working?

File: ui/src/app/pages/pages-routing.module.ts [hunk 5]
```diff
@@ -1,234 +1,135 @@
-import {RouterModule, Routes} from '@angular/router';
```
Comment (daitan-rovaris) [inline ui/src/app/pages/pages-routing.module.ts:1]: Please, in the routes check if these still exist, and makes sensor for the ORB project, if not please remove them. also, this logic where we have a few routes for production and other ones for development, won't be necessary for the time being.

File: ui/src/app/pages/pages.module.ts [hunk 6]
```diff
@@ -1,17 +1,16 @@
 import { NgModule } from '@angular/core';
```
Comment (daitan-rovaris) [inline ui/src/app/pages/pages.module.ts:1]: Please remove all unused modules for orb, like the twins page and so forth;

File: ui/src/app/pages/sinks/add/sinks.add.component.html [hunk 7]
```diff
@@ -1,35 +1,60 @@
...
   <nb-card-header class="table-header">
-    {{this.action}} Sink
+    {{ this.action }} New Sink
   </nb-card-header>
-
   <nb-card-body>
     <div class="row form-row">
       <div class="col-3 form-key">
-        Name:
+        Name<!-- TODO add red asterisk here with a span or something-->
```
Comment (daitan-rovaris) [inline ui/src/app/pages/sinks/add/sinks.add.component.html:8]: Please remove the TODO

**Review by daitan-rovaris — CHANGES_REQUESTED** [review]
Overall looks great to me, I understand that this is the first big PR, and there are a lot of assets being shipped, and styling rules being sorted out but, let's try keeping PR's at a minimum in terms of lines of code.

File: ui/package.json [hunk 8]
```diff
@@ -13,6 +13,7 @@
     "ng": "ng",
     "conventional-changelog": "conventional-changelog",
     "start": "ng serve --proxy-config=proxy-config.json",
+    "start:proxydocker": "ng serve --proxy-config=proxy-config-docker.json",
```
[the remark this answers is not in the document]
Comment (gpazuch) [reply ui/package.json]: 'start:proxydocker' and related files have been removed

to build and serve ui locally during development, run '$>npm run start' or 'yarn start'

File: ui/src/app/common/services/sinks/sinks.service.ts (hunk 3 above)
Comment (gpazuch) [reply ui/src/app/common/services/sinks/sinks.service.ts:12]: This constants are defined everywhere in mainflux codebase following this camelCase style.

I'll try and update these together with the lint fixes.

File: ui/src/app/pages/agents/agents.component.html (hunk 4 above)
Comment (gpazuch) [reply ui/src/app/pages/agents/agents.component.html]: This has been resolved.

**Review by daitan-rovaris — APPROVED** [review]
lgtm

**Pull request merged**